db/migrate/20200922230337_create_services.rb

```diff
@@ -0,0 +1,31 @@
+class CreateServices < ActiveRecord::Migration[4.2]
+  def self.up
+    create_table :services, id: :bigint do |t|
+      t.string :name, :null => false, :limit => 255, :required => true
+      t.string :ref, :null => false, :limit => 255, :required => true, :default => "master"
+    end
+    add_index :services, [:name], :unique => true, :name => 'on_name'
+
+    add_column :pushes, :service_id, :integer, :limit => 8, :null => false
+
+    add_index :pushes, [:service_id], :name => 'on_service_id'
+
+    web_service = Service.create!(name: 'web', ref: 'production')
+    Service.create!(
+      [
+        { name: 'rs_west',    ref: '2b7a8338fde0a998d8aa6b540f1aa4dcb3f9018f' },
+        { name: 'rs_east',    ref: '2b7a8338fde0a998d8aa6b540f1aa4dcb3f9018f' },
+        { name: 'rs_central', ref: '2b7a8338fde0a998d8aa6b540f1aa4dcb3f9018f' }
+      ]
+    )
+    Push.update_all(service: web_service)
```

Comment:
Small thing here, I'm not sure if Rails' `update_all` is clever enough to handle being given an object that should be set as a foreign key ID column value (`service_id` on the `services` table).

You could test it in a local console session to confirm. If it doesn't work we could change this to `Push.update_all(service_id: web_service.id)`

Reply:
wow great catch! the way I had it didn't work.. I updated to use `Push.update_all(service_id: web_service.id)` after confirming that method works